R/teal_data-get_code.R

```diff
@@ -103,23 +105,22 @@
 #' @aliases get_code,teal_data-method
 #'
 #' @export
-setMethod("get_code", signature = "teal_data", definition = function(object, deparse = TRUE, datanames = NULL, ...) {
-  checkmate::assert_character(datanames, min.len = 1L, null.ok = TRUE)
-  checkmate::assert_flag(deparse)
-
-  code <- if (!is.null(datanames)) {
-    get_code_dependency(object@code, datanames, ...)
-  } else {
-    object@code
-  }
+setMethod("get_code",
+  signature = "teal_data",
+  definition = function(object, deparse = TRUE, names = NULL, datanames = lifecycle::deprecated(), ...) {
+    checkmate::assert_character(names, min.len = 1L, null.ok = TRUE)
+    checkmate::assert_flag(deparse)
 
-  if (deparse) {
-    if (length(code) == 0) {
-      code
-    } else {
-      paste(code, collapse = "\n")
+    if (lifecycle::is_present(datanames)) {
+      lifecycle::deprecate_warn(
+        when = "0.6.1",
+        what = "teal.data::get_code(datanames = )",
+        with = "teal.code::get_code(names = )",
+        always = TRUE
+      )
+      checkmate::assert_character(datanames, min.len = 1L, null.ok = TRUE)
```

Comment:
if you move that if before assert then there won't be a need to repeat the logic

Reply:
https://github.com/insightsengineering/teal.data/pull/343/commits/4ce5c54eac9a97bfbde9513d6eab31ae98b582e9